gateway-js/src/index.ts

```diff
@@ -351,24 +430,41 @@ export class ApolloGateway implements GraphQLService {
       }
     }
 
-    await this.updateComposition();
-    if (
-      (isManagedConfig(this.config) || this.experimental_pollInterval) &&
-      !this.pollingTimer
-    ) {
-      this.pollServices();
-    }
+    const result = isStaticConfig(this.config)
+      ? this.loadStatic()
+      : await this.loadDynamic();
 
     const mode = isManagedConfig(this.config) ? 'managed' : 'unmanaged';
-
     this.logger.info(
       `Gateway successfully loaded schema.\n\t* Mode: ${mode}${
-        (this.apolloConfig && this.apolloConfig.graphId)
+        this.apolloConfig && this.apolloConfig.graphId
           ? `\n\t* Service: ${this.apolloConfig.graphId}@${this.apolloConfig.graphVariant}`
           : ''
       }`,
     );
 
+    return result;
+  }
+
+  private loadStatic() {
+    this.maybeWarnOnConflictingConfig();
+
+    return {
```

Comment:
This is identical across static and dynamic, so maybe this isn't necessary and can just be returned from `load`?

Maybe even just eliminate this function and inline the maybeWarn into the caller? Actually can we just call maybeWarn in one place, if !isManagedConfig?

Reply:
Yeah, this is a good suggestion.
525df37